commerce/bookstore/python/load_data.py

```diff
@@ -0,0 +1,286 @@
+
+    return TypeQL_insert_query
+
+
+def orders_template(order):  # building a TypeQL request to insert an order
+    TypeQL_insert_query = 'insert $o isa Order, has id "' + order["id"] + '",' \
+                          'has foreign-user-id "' + order["User-ID"] + '", ' \
+                          'has date ' + order["date"] + ', ' \
+                          'has status "' + order["status"] + '",' \
+                          'has delivery_address "' + order["delivery_address"] + '", ' \
+                          'has payment_details "' + order["payment_details"] + '";'
+
+    return TypeQL_insert_query
+
+
+def generate_ordered_items():  # Generating item-lists for orders from books
+    result = []
+    # generate 5 random sets of 2-9 books
+    with TypeDB.core_client("localhost:1729") as client:
+        with client.session(db, SessionType.DATA) as session:
+            with session.transaction(TransactionType.READ) as transaction:
+                TypeQL_read_query = 'match $b isa Book, has ISBN $x; get $x; limit 800;'
+                #print("Executing TypeQL read Query: " + TypeQL_read_query)
+                iterator = transaction.query().match(TypeQL_read_query)
+                answers = [ans.get("x") for ans in iterator]
+                books = [answer.get_value() for answer in answers]
+                for order_id in range(1,6):
+                    ordered_books = []
+                    for item_n in range(1, random.randint(2, 10)):
+                        ordered_books.append(books[random.randint(0, 799)])  # Exactly 800 books to select from
+                    result.append(ordered_books)
+
+    n = 1
+    # Add ordering relations (assign items from the sets above)
+    with TypeDB.core_client("localhost:1729") as client:
+        with client.session(db, SessionType.DATA) as session:
+            for order in result:
+                #print('\nOrder #', n, 'contains:')
+                for book in order:
+                    #print('\nISBN', book)
+                    with session.transaction(TransactionType.WRITE) as transaction:
+                        TypeQL_insert_query = 'match $b isa Book, has ISBN "' + book + '";' \
+                                              '$o isa Order, has id "' + str(n) + '", has foreign-user-id $fui;' \
+                                              '$u isa User, has foreign-id $fi;' \
+                                              '$fui = $fi;' \
+                                              'insert (order: $o, item: $b, author: $u ) isa ordering;'
+                        # the $fui and $fi variables are compared by value only
+                        #print("Executing TypeQL Query: " + TypeQL_insert_query)
+                        transaction.query().insert(TypeQL_insert_query)
+                        transaction.commit()
+                n += 1
+    return  # END of generate_ordered_items()
+
+
+def load_genre_tags():  # Creating genre tags and tag hierarchy
```

Comment:
This method looks very "un-code-like", in that we insert a bunch of data that we've hardcoded into Python, as opposed to encoding it in the source data (`genres.csv`, or elsewhere).

I assume that's because the source data has no notion of a tag _hierarchy_, and so we've had to fabricate one, and it was convenient to just insert the genres themselves in the same method.

So I would do two things (at least!) here to make this function read and behave more like what a developer would expect from a function about loading genre tags.

1. Split the function up into two functions: one for loading the `genre` attributes, one for creating the hierarchies of genres.
2. Structure the `genre` attribute loader function the same way as the

Reply:
1. I would prefer to leave it just like that. As it shows a few important teachable moments here. First - we have a lot of requests going on with a single commit. That is an alternative to other request cases we have in this example. Second - we have genre hierarchy relation requests close to the involved genre requests. That's kinda easier to grasp, then in separate functions. 
P.S. Yeah, I hard-coded genres because that's something we wouldn't want to load as initial data. That's why I was looking for a way to implement tags with schema before. This way genres are not in the schema but also not a part of loaded data IMHO. I think I can change that hard-coded behaviour to loading 1 or 2 `.tql` files if you want.

Reply:
Let's just rename this method to be clear that it is fabricating data, as opposed to loading data from CSV (which most of the rest of this file is doing). Something like `generate_tag_hierarchy` or `synthesise_genre_data` ... etc etc

Reply:
create_genre_tags() now - https://github.com/vaticle/typedb-examples/pull/143/commits/6d50dec3400da9693ed442714a99918bf2080987